index.js

```diff
@@ -119,16 +99,80 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:((?!\.|_)@?[0-9a-zA-Z./_%-]{1,214}):)?([^]+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (Nullable String, String, Number)
+  //                      -> TypeIdentifier
+  function TypeIdentifier(namespace, name, version) {
+    return {
+      namespace: namespace,
+      name: name,
+      version: version
+    };
+  }
+
+  //       parseTypeIdentifier :: String -> TypeIdentifier
+  function parseTypeIdentifier(s) {
+    if (s === '') return TypeIdentifier(null, s, 1);
+    var parsed = RPARSE.exec(s);
```

Comment:
I had this when the regular expression did not allow for new lines. Now however, it will consume any kind of string input of at least one character. I changed it to this to be more explicit. I'll change it, but it will have no functional impact. I suppose the function will become more resistant to future changes.